citadel/indico_citadel/backend.py

```diff
@@ -99,28 +100,23 @@ def upload_record(self, entry, entries, session, from_queue):
             raise Exception(f'{response.status_code} - {response.text} in record {json}')
         elif change_type & SimpleChange.created:
             content = response.json()
-            control_number = content['metadata']['control_number']
-            if control_number:
-                CitadelSearchAppIdMap.create(control_number, entry.id, entry_type)
-            else:
-                raise Exception('Cannot create the search id mapping: {} - {}'
-                                .format(response.status_code, response.text))
+            CitadelSearchAppIdMap.create(content['metadata']['control_number'], entry.id, entry_type)
         response.close()
 
     def upload_file(self, entry, entries, session):
         with entry.attachment.file.open() as file:
             response = session.put(
-                url_join(self.search_app, f'api/record/{entry.search_id}/files/{entry.attachment.file.filename}'),
+                url_join(self.search_app, f'api/record/{entry.search_id}/files/attachment'),
                 data=file
             )
         if response.ok:
             entry.attachment_file_id = entry.attachment.file.id
             db.session.merge(entry)
             db.session.commit()
         else:
-            self.logger.error('Failed uploading attachment %d', entry.attachment.id)
+            self.logger.error('Failed uploading attachment %d: %s', entry.attachment.id, response.text)
 
-    def run_initial(self, events, total=None):
+    def run_initial(self, events, total=None, progress=True):
         cte = Category.get_tree_cte(lambda cat: db.func.json_build_object('id', cat.id, 'title', cat.title))
         self.categories = dict(db.session.execute(select([cte.c.id, cte.c.path])).fetchall())
         super().run_initial(events, total)
```

Comment:
shouldn't we pass `progress` here as well?

Reply:
TBH I'm not a fan of this progress flag as it seems redundant. 
I remember we discussed this before. What I would do is disable the progress output if `total` is zero/None. Or actually patching the `verbose_iterator` if it feels more strict to always ensure a total.

Reply:
ah right we discussed that - i don't remember if we decided on the flag or on patching `verbose_iterator` for tests (the latter is probably cleaner)... i just commented because of the inconsistency here.